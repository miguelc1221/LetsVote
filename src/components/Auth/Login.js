import React, { Component, PropTypes } from "react";
import { Button, Menu, Form, Modal, Message } from "semantic-ui-react";
import { connect } from "react-redux";
import * as authActions from "../../actions/auth";
import "./Auth.css";

class Login extends Component {
  static propTypes = {
    error: PropTypes.object,
    isFetching: PropTypes.bool
  };

  constructor() {
    super();

    this.state = {
      email: "",
      password: "",
      modalOpen: false
    };
    this.handleOnSubmit = this.handleOnSubmit.bind(this);
    this.handleOnChange = this.handleOnChange.bind(this);
    this.handleOnOpen = this.handleOnOpen.bind(this);
    this.handleOnClose = this.handleOnClose.bind(this);
  }

  handleOnChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  handleOnSubmit(e) {
    e.preventDefault();
    this.props.loginUser(this.state);
  }

  handleOnOpen() {
    this.setState({ modalOpen: true });
  }

  handleOnClose() {
    this.props.clearErrors();
    this.setState({ modalOpen: false });
  }

  render() {
    console.log(this.state);
    const { email, password } = this.state;
    const { loginError, isFetching } = this.props.auth;
    return (
      <Modal
        className="auth"
        size="small"
        trigger={
          <Menu.Item
            position="right"
            name="Login"
            onClick={this.handleOnOpen}
          />
        }
        open={this.state.modalOpen}
        onClose={this.handleOnClose}
      >
        <Modal.Header className="auth__header">Login</Modal.Header>
        <Modal.Content>
          <Form onSubmit={this.handleOnSubmit} error={!!loginError}>
            <Form.Field>
              <label>Email</label>
              <input
                placeholder="Email"
                name="email"
                onChange={this.handleOnChange}
              />
            </Form.Field>
            <Form.Field>
              <label>Password</label>
              <input
                type="password"
                name="password"
                placeholder="Password"
                onChange={this.handleOnChange}
              />
              <Message
                error
                hidden={!loginError.Error}
                content={loginError.Error ? loginError.Error : ""}
              />
            </Form.Field>
            <Button
              color="olive"
              type="submit"
              disabled={!email || !password}
              loading={isFetching}
            >
              Login
            </Button>
          </Form>
        </Modal.Content>
      </Modal>
    );
  }
}

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(mapStateToProps, authActions)(Login);
