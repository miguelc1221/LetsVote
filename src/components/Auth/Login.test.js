import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Login } from './Login';

describe('Login Component', () => {
  it('should render as expected', () => {
    const component = shallow(<Login auth={{loginError: {}}} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  })
})