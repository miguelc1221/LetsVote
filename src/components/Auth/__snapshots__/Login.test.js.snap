exports[`Login Component should render as expected 1`] = `
<Modal
  className="auth"
  closeIcon="close"
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  dimmer={true}
  onClose={[Function]}
  open={false}
  size="small"
  trigger={
    <MenuItem
      name="Login"
      onClick={[Function]}
      position="right" />
  }>
  <ModalHeader
    className="auth__header">
    Login
  </ModalHeader>
  <ModalContent>
    <Form
      as="form"
      error={true}
      onSubmit={[Function]}>
      <FormField>
        <label>
          Email
        </label>
        <input
          name="email"
          onChange={[Function]}
          placeholder="Email" />
      </FormField>
      <FormField>
        <label>
          Password
        </label>
        <input
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password" />
        <Message
          content=""
          error={true}
          hidden={true} />
      </FormField>
      <Button
        as="button"
        color="olive"
        disabled={true}
        type="submit">
        Login
      </Button>
    </Form>
  </ModalContent>
</Modal>
`;
